java/arcs/android/sdk/host/IntentArcHostAdapter.kt

```diff
@@ -76,17 +94,29 @@ class IntentArcHostAdapter(
 
     @OptIn(ExperimentalCoroutinesApi::class)
     class ResultReceiverContinuation<T>(
-        val continuation: CancellableContinuation<T?>,
+        val continuation: Continuation<T?>,
         val block: (Any?) -> T?
     ) : ResultReceiver(Handler()) {
-        override fun onReceiveResult(resultCode: Int, resultData: Bundle?) =
+        override fun onReceiveResult(resultCode: Int, resultData: Bundle?) {
+            val exception = resultData?.getString(ArcHostHelper.EXTRA_OPERATION_EXCEPTION)
+            if (exception != null) {
+                continuation.resumeWithException(
+                    ArcHostException(
+                        exception,
+                        resultData.getString(ArcHostHelper.EXTRA_OPERATION_EXCEPTION_STACKTRACE)
+                            ?: ""
```

Comment:
I think there's an overload for this method that lets you specify a default value